<script type="text/javascript" src="/duplication/scripts/jhome.js"></script>
<p>
PGDD is a public database to identify and catalog plant genes in terms of intragenome or cross-genome syntenic relationships. Current efforts focus on flowering plants with available whole genome sequences (preferrably assembled pseudomolecules with ordered gene models). 
</p>
<h1>Data sources</h1>
<table class="sofT" cellspacing="0">
<tr class="helpHed">
	<td colspan="7">Plant genomes in this database (10 genomes)</td>
</tr>
<tr class="helpHed">
	<td colspan="2">Species name</td>
    <td>Common name</td>
	<td>Release version</td>
	<td>Gene number</td>
	<td>Access</td>
	<td>Reference</td>
</tr>
<tr>
    <td><img src="/duplication/images/icons/at.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=3702"><i>Arabidopsis thaliana</i></a></td>
    <td>thale cress</td>
	<td>TAIR 8.0 (Aug. 2008)</td>
	<td>26990</td>
	<td><a href="ftp://ftp.arabidopsis.org/home/tair/Genes/">TAIR</a></td>
    <td><a href="http://www.nature.com/nature/journal/v408/n6814/abs/408796a0.html"><i>Nature</i></a></td>
</tr> 
<tr>
    <td><img src="/duplication/images/icons/cp.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=3649"><i>Carica papaya</i></a></td>
        <td>papaya</td>
        <td>EVM (Jul. 2007)</td>
        <td>25536</td>
        <td><a href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?db=nucleotide&amp;val=186167268">Genbank</a>, <a href="ftp://asgpb.mhpcc.hawaii.edu/papaya/">Hawaii</a></td>
        <td><a href="http://www.nature.com/nature/journal/v452/n7190/abs/nature06856.html"><i>Nature</i></a></td>
</tr>
<tr>
    <td><img src="/duplication/images/icons/pt.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=3694"><i>Populus trichocarpa</i></a></td>
    <td>poplar</td>
    <td>JGI 2.0 (Feb. 2010)</td>
	<td>45778</td>
	<td><a href="ftp://ftp.jgi-psf.org/pub/JGI_data/phytozome/v5.0/Ptrichocarpa/">JGI</a></td>
    <td><a href="http://www.sciencemag.org/cgi/content/abstract/313/5793/1596"><i>Science</i></a></td>
</tr>
<tr>
    <td><img src="/duplication/images/icons/mt.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=3880"><i>Medicago trunculata *</i></a></td>
    <td>barrel medic</td>
	<td>Release 2.0 (Feb. 2008)</td>
	<td>38834</td>
    <td><a href="http://www.medicago.org/genome/downloads/Mt2/">UMN</a></td>
    <td>--</td>
</tr>
<tr>
    <td><img src="/duplication/images/icons/gm.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=3847"><i>Glycine max</i></a></td>
        <td>soybean</td>
        <td>Release 1 (Dec. 2008)</td>
        <td>75778</td>
        <td><a href="ftp://ftp.jgi-psf.org/pub/JGI_data/Glycine_max/">JGI</a></td>
        <td><a href="http://www.nature.com/nature/journal/v463/n7278/abs/nature08670.html"><i>Nature</i></a></td>
</tr>
<tr>
    <td><img src="/duplication/images/icons/vv.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=29760"><i>Vitis vinifera</i></a></td>
        <td>grape</td>
        <td>Genoscope (Aug. 2007)</td>
        <td>30434</td>
        <td><a href="http://www.genoscope.cns.fr/externe/GenomeBrowser/Vitis/">Genoscope</a></td>
        <td><a href="http://www.nature.com/nature/journal/v449/n7161/abs/nature06148.html"><i>Nature</i></a></td>
</tr>
<tr>
    <td><img src="/duplication/images/icons/bd.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=15368"><i>Brachypodium distachyon</i></a></td>
        <td>purple false brome</td>
        <td>Release (May 2009)</td>
        <td>32255</td>
        <td><a href="ftp://ftp.jgi-psf.org/pub/JGI_data/phytozome/v4.0/">JGI</a></td>
        <td><a href="http://www.nature.com/nature/journal/v463/n7282/full/nature08747.html"><i>Nature</i></a></td>
</tr>
<tr> 
    <td><img src="/duplication/images/icons/os.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=4530"><i>Oryza sativa</i></a></td>
    <td>rice</td>
	<td>RAP 2.0 (Nov. 2007)</td>
	<td>30192</td>
    <td><a href="http://rapdb.dna.affrc.go.jp/">RAP</a></td>
    <td><a href="http://www.nature.com/nature/journal/v436/n7052/full/nature03895.html"><i>Nature</i></a></td>
</tr>
<tr>
    <td><img src="/duplication/images/icons/sb.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=4558"><i>Sorghum bicolor</i></a></td>
        <td>sorghum</td>
        <td>Sbi 1.4 (Dec. 2007)</td>
        <td>34496</td>
        <td><a href="http://genome.jgi-psf.org/Sorbi1/">JGI</a></td>
        <td><a href="http://www.nature.com/nature/journal/v457/n7229/full/nature07723.html"><i>Nature</i></a></td>
    </tr>
<tr>
    <td><img src="/duplication/images/icons/zm.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=4577"><i>Zea mays</i></a></td>
        <td>corn</td>
        <td>Release 4a.53 (Oct. 2009)</td>
        <td>32540</td>
        <td><a href="http://ftp.maizesequence.org/">AGI</a></td>
        <td><a href="http://www.sciencemag.org/cgi/content/full/326/5956/1112"><i>Science</i></a></td>
</tr>

    <tr class="helpHed"><td colspan="7">In queue (5 genomes)</td></tr>
<tr>
    <td><img src="/duplication/images/icons/al.png" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=59689"><i>Arabidopsis lyrata *</i></a></td>
        <td>lyrate rockcress</td>
        <td>Release (Jul. 2008)</td>
        <td>--</td>
        <td><a href="ftp://ftp.jgi-psf.org/pub/JGI_data/phytozome/v4.0/">JGI</a></td>
        <td>--</td>
    </tr>
<tr>
    <td><img src="/duplication/images/icons/cs.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=3659"><i>Cucumis sativus</i></a></td>
        <td>cucumber</td>
        <td>Release (Oct. 2009)</td>
        <td>--</td>
        <td><a href="ftp://cucumber.genomics.org.cn/pub/cucumber/">BGI</a></td>
        <td><a href="http://www.nature.com/ng/journal/vaop/ncurrent/abs/ng.475.html"><i>Nature Genetics</i></a></td>
</tr>
<tr>
    <td><img src="/duplication/images/icons/rc.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=3988"><i>Ricinus communis *</i></a></td>
        <td>castor bean</td>
        <td>Release 0.1 (May 2008)</td>
        <td>38613</td>
        <td><a href="http://castorbean.tigr.org/castorbean_downloads.shtml">TIGR</a></td>
        <td>--</td>
</tr>
<tr>
    <td><img src="/duplication/images/icons/lj.gif" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=34305"><i>Lotus japonicus</i></a></td>
        <td>lotus</td>
        <td>Release (May 2009)</td>
        <td>42399</td>
        <td><a href="ftp://ftp.kazusa.or.jp/pub/lotus/">Kazusa</a></td>
        <td><a href="http://dnaresearch.oxfordjournals.org/cgi/content/abstract/dsn008v1"><i>DNA research</i></a></td>
</tr>
<tr>
    <td><img src="/duplication/images/icons/mg.png" width="25" height="25" alt="" /></td>
    <td><a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=4155"><i>Mimulus guttatus *</i></a></td>
        <td>monkey flower</td>
        <td>Release (Jan 2010)</td>
        <td> -- </td>
        <td><a href="ftp://ftp.jgi-psf.org/pub/JGI_data/phytozome/v5.0/Mguttatus">JGI</a></td>
        <td> -- </td>
</tr>
</table>
<br />
<li>* Un-published genome data temporarily restricted for downloading (in accordance with the understandings in the <a href="http://www.genome.gov/Pages/Research/WellcomeReport0303.pdf">Fort Lauderdale meeting</a> and <a href="http://www.genome.gov/page.cfm?pageID=10506537">NHGRI policy statement</a>)</li>
<li>Processed sequences and coordinates for the gene models are available <a href="/duplication/mcscan/index/results">here</a>.</li>
<li><a href="http://www.phytozome.net/Phytozome_info.php">Phytozome</a> info page contains current repository of plant genome data. </li> 
<p><b>The duplication history of major angiosperm taxa</b><br />
<i>lavender circles represent inferred polyploidy events, drawn roughly to scale</i></p>
<img src="/duplication/images/organism.png" class="articleimg" id="tree" width="500" alt="" />
<br />
<h1>Methods</h1>
<div onclick="$('#p1').toggle();"><h3><img src="/duplication/images/expand.png" width="20" height="20" alt="" /> Identify syntenic blocks</h3></div>
<p id="p1" style="display: none">
We used BLASTP to search for potential anchors (<i>E</i> &lt;1e-5, top 5 matches) between every possible pair of chromosomes in multiple genomes. The homologous pairs are used as the input for <a href="/duplication/mcscan">MCscan</a>. MCscan is a novel synteny search program that combines the merits of two existent algorithms. The built-in scoring scheme for MCscan is <i>min</i> {-log<sub>10</sub><i>E</i>, 50} for every matching gene pairs and -1 for each 10kb distance between anchors, similar to <a href="http://dagchainer.sourceforge.net/">DAGchainer</a> and blocks that have scores &gt;300 were kept. The resulting syntenic chains are evaluated using a procedure in <a href="http://colinear.cbi.pku.edu.cn/">ColinearScan</a> and <i>E</i>-value &lt;1e-10 were used as a significance cutoff.
</p>
<div onclick="$('#p2').toggle();"><h3><img src='/duplication/images/expand.png' width="20" height="20" alt="" /> Calculate synonymous substitutions (Ks)</h3></div>
<p id='p2' style="display: none">
For homologs inferred from syntenic alignments, we aligned the protein sequences of the gene pairs using <a href="http://www.ebi.ac.uk/Tools/clustalw2/index.html">CLUSTALW</a> and used the protein alignments to guide CDS alignments by <a href="http://coot.embl.de/pal2nal/">PAL2NAL</a>. Finally, we used Nei-Gojobori method implemented in the <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">PAML</a> package to calculate Ks. An in-house <a href="/duplication/data/syn_calc.zip">python script</a> is used to pipeline all the calculations. Log-gaussian mixture models are fitted to the Ks distributions using <a href="http://astro.u-strasbg.fr/~fmurtagh/mda-sw/">GMM with Bayes Factors</a>.
</p>
<div onclick="$('#p3').toggle();"><h3><img src='/duplication/images/expand.png' width="20" height="20" alt="" /> How can I calculate the significance of segmental duplication</h3></div>
<p id='p3' style="display: none">
Number of collinear genes: <input type="text" size="5" id="K" />
Total number of genes in genome: <input type="text" size="5" id="N" /><br />
Spread in location A: <input type="text" size="5" id="J1"> 
Spread in location B: <input type="text" size="5" id="J2"><br />
<button onclick="calc();">Calculate</button> 
<button onclick="ex();">Show me an example</button>
</p>
<div class="monotext" id="display" style="display: none"></div>
<br />
<h1>How to cite</h1>
<ul>
<li><b>Tang, H., Bowers, J.E., Wang, X., Ming, R., Alam, M. and Paterson, A.H.</b> (2008) <i>Synteny and Collinearity in Plant Genomes</i>, Science, 320, 486-488. [ <a href="http://www.sciencemag.org/cgi/content/full/sci;320/5875/486">Online</a> ]</li>
<li><b>Tang, H., Wang, X., Bowers, J.E., Ming, R., Alam, M. and Paterson, A.H.</b> (2008) <i>Unraveling ancient hexaploidy through multiply-aligned angiosperm gene maps</i>, Genome Research, 18, 1944-1954. [ <a href="http://genome.cshlp.org/content/18/12/1944.full">Online</a> ] </li>
</ul>
<br /><br />
